pydis_site/apps/api/viewsets/bot/infraction.py

```diff
@@ -133,10 +134,27 @@ class InfractionViewSet(
 
     serializer_class = InfractionSerializer
     queryset = Infraction.objects.all()
+    pagination_class = LimitOffsetPagination
     filter_backends = (DjangoFilterBackend, SearchFilter, OrderingFilter)
     filter_fields = ('user__id', 'actor__id', 'active', 'hidden', 'type')
     search_fields = ('$reason',)
     frozen_fields = ('id', 'inserted_at', 'type', 'user', 'actor', 'hidden')
+    LimitOffsetPagination.default_limit = 100
+
+    def list(self, request: HttpRequest, *args, **kwargs) -> Response:
+        """
+        DRF method for listing Infraction entries.
+
+        Called by the Django Rest Framework in response to the corresponding HTTP request.
+        """
+        queryset = self.filter_queryset(self.get_queryset())
+        page = self.paginate_queryset(queryset)
+        if page:
+            serializer = self.get_serializer(page, many=True)
+            result = self.get_paginated_response(serializer.data)
+            return Response(result.data.get('results'))
```

Comment:
If I understand correctly, you overwrote the `list` method just to customise the response. Specifically, you want to only return the raw data from the paginated response and not any of the pagination metadata. If that's the case, a better way is to [create a custom paginator](https://www.django-rest-framework.org/api-guide/pagination/#custom-pagination-styles) which overwrites `get_paginated_response`. This approach achieves the same result with less redundant code since it can rely on the `list` implementation provided by the mixin.

In any case, why is this being done? Is it to maintain the old API response format? How is the client supposed to paginate without knowing the final page?

Reply:
Yes, the sole purpose to override this method is to maintain the old response.
If the response has been changed then any other services which is using the older response will break. So before updating the response we'll need to update those services as well.

Reply:
They'd have to be updated anyway unless it is decided that providing more than the 100 most recent infractions for any query is unnecessary.

Reply:
So do I need to send updated response with pagination metadata?

Reply:
That is contingent on what moderators decide they need. You'll have to wait for a response from them.

Reply:
Ok

Reply:
Any update on this?

Reply:
I believe it was decided a single page is fine, and we can maintain the old result format. Mark's other comment on custom paginators is still unresolved though.

Reply:
Cool then I'll work on the custom paginator.

Reply:
Please update the documentation on the endpoint too. It should mention that the results are paginated but only the actual data is returned.

Reply:
Changes have been done and pushed, take a look at the new commit.